### DISCUSSION TITLE
Overwrite Config in SubClass to have `anystr_lower = True` return all lowercase values

### DISCUSSION THREAD

**Discussion by matthewdchambers** [pydantic/pydantic#3333] [Question] [answered]
Hey gang- excellent package.

Attempting to set all strings to lowercase in a SubClass, but not getting the expected behaviour. 

```
from pydantic import BaseModel

class Foo(BaseModel):
    foo: str

class Bar(Foo):
    bar: str

    class Config:
        anystr_lower = True


f = Foo(foo='FOO')
b = Bar(foo='FOO', bar='BAR')

print(f.dict())
print(b.dict())
```

Yields: 
```
{'foo': 'FOO'}
{'foo': 'FOO', 'bar': 'bar'}
```

While I was expecting: 
```
{'foo': 'FOO'}
{'foo': 'foo', 'bar': 'bar'}
```

Is there a way to have the `Config` specification of a subclass apply to all values in the parent, or to otherwise get the desired result? 

Thanks!

**Comment by Kludex** [comment, marked as answer]
I don't know when this was fixed, but on V2, you have the behavior that you describe.

```py
from pydantic import BaseModel, ConfigDict


class Foo(BaseModel):
    foo: str


class Bar(Foo):
    model_config = ConfigDict(str_to_lower=True)

    bar: str


f = Foo(foo="FOO")
b = Bar(foo="FOO", bar="BAR")

print(f.model_dump())
print(b.model_dump())
```

Notes:
- `model_config` replaced `Config` class on V2.
- `model_dump` replaced `dict` on V2.
